backend/main/shared/flow.go

```diff
@@ -181,6 +196,57 @@ func (fa *FlowAdapter) UserTransactionValidate(address string, message string, s
 	return nil
 }
 
+func (fa *FlowAdapter) EnforceTokenThreshold(creatorAddr string, c *Contract) (bool, error) {
+
+	flowAddress := flow.HexToAddress(creatorAddr)
+	cadenceAddress := cadence.NewAddress(flowAddress)
+	cadencePath := cadence.Path{Domain: "public", Identifier: *c.Public_path}
+
+	script, err := ioutil.ReadFile("./main/cadence/get_balance.cdc")
+	if err != nil {
+		log.Error().Err(err).Msgf("error reading cadence script file")
+		return false, err
+	}
+
+	script = replaceContractPlaceholders(string(script[:]), c)
+
+	//call the script to verify balance
+	cadenceValue, err := fa.Client.ExecuteScriptAtLatestBlock(
+		fa.Context,
+		script,
+		[]cadence.Value{
+			cadencePath,
+			cadenceAddress,
+		},
+	)
+	if err != nil {
+		log.Error().Err(err).Msg("error executing script")
+		return false, err
+	}
+
+	value := CadenceValueToInterface(cadenceValue)
+	balance, err := strconv.ParseFloat(value.(string), 64)
+	if err != nil {
+		log.Error().Err(err).Msg("error converting cadence value to float")
+		return false, err
+	}
+
+	if balance < *c.Threshold {
+		return false, errors.New("balance is below threshold")
+	}
+
+	return true, nil
+}
+
+// Fungible Token Address here is hardcoded to emulator address, this should
+// be set based on environment
+func replaceContractPlaceholders(code string, c *Contract) []byte {
+	code = placeholderFungibleTokenAddr.ReplaceAllString(code, "0xee82856bf20e2aa6")
```

Comment:
should remove the hardcoded address and replace with environment variable.

Reply:
we should probably use the `flow.json` to define and fetch the contract address mappings based on environment.

... i'll brainstorm the best way to do this